### PR TITLE
Coffeescript to js conversions

### PR DESCRIPTION
**Pull request by Page-**
Resolves: # 
HQ: <url> 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/logs.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import { EventEmitter } from 'events';
+import * as ndjson from 'ndjson';
+import { AbortController as AbortControllerPonyfill } from 'abortcontroller-polyfill/dist/cjs-ponyfill';
+import { findCallback } from './util';
+import { globalEnv } from './util/global-env';
+
+const AbortController =
+	globalEnv != null && 'AbortController' in globalEnv
+		? globalEnv.AbortController
+		: AbortControllerPonyfill;
```
Comment (Page-) [inline lib/logs.js:28]: This changed slightly to make typescript happy but the effect should be equivalent, previous was
```coffee
AbortController = globalEnv?.AbortController || AbortControllerPonyfill

File: lib/settings.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as Promise from 'bluebird';
+
+const getSettings = function(deps) {
```
Comment (Page-) [inline lib/settings.js:19]: Removed an unused `opts` parameter which should be safe to do

File: lib/logs.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+				baseUrl: opts.apiUrl,
+			})
+			.get('body');
+	};
+
+	const subscribeToApiLogs = function(device, options) {
+		if (options == null) {
+			options = {};
+		}
+		/** @type {LogSubscription & { unsubscribe?: () => void } } */
```
Comment (thgreasi) [inline lib/logs.js:58]: I think that this broke the docs builder.

Comment (Page-) [reply lib/logs.js]: Turns out just turning it into a normal comment does the trick

**Comment by Page-** [comment]
@balena-ci retest

**Pull request merged**